oem/ibm/libpldmresponder/utils.cpp

```diff
@@ -604,6 +604,33 @@ void findSlotObjects(const std::string& boardObjPath,
     }
 }
 
+std::vector<char> vecSplit(const std::vector<char>& inputVec,
+                           const uint32_t startIdx, const uint32_t endIdx)
+{
+    // Start and end iterators
+    auto startItr = inputVec.begin() + startIdx;
+    auto endItr = inputVec.begin() + endIdx;
+
+    // Resultant split vector
+    std::vector<char> resultVec(endIdx - startIdx + 1);
+
+    copy(startItr, endItr, resultVec.begin());
+    return resultVec;
+}
+
+uint32_t readVecContent(std::vector<char>& inputVec, const uint32_t startIdx,
+                        const uint32_t endIdx)
+{
+    uint32_t size = 0;
+    std::vector<char> userPassLenArr(inputVec.begin() + startIdx,
+                                     inputVec.begin() + endIdx);
+    for (uint32_t idx = 0; idx < 8; idx++)
+    {
+        size |= (uint32_t)userPassLenArr[idx] << 8 * idx;
+    }
+    return size;
+}
```

Comment:
range based for loop , and magic number 8 , and this cannot be a util function since we have decided that it takes a vector<char>& , uint32 start index & uint32_t endindex and returns uint32_t number.

util API's should be really be scalable and handle multiple types e.t.c i dont see any other part of code re-using this function in future. do you forsee any code that could re-use this function ? if not this is not a util function.

Reply:
and there are also some backed assumptions in this function like for example, the function assumes that inputVec contains at least endIdx elements and that the subrange specified by startIdx and endIdx contains at least 8 bytes. Otherwise, the behavior may be undefined. Additionally, it assumes the little-endian byte order for interpreting the bytes as a 32-bit integer. A util API should not have any assumptions implicitly. I suggest to not have this function , instead figure out anyother way to extract the size directly.

Reply:
I could not find a better working alternative to this. I have removed the magic number and moved this out of util and put into SmsHandler itself. Since I need to the upper limit value not able to move to range based loop. Will revisit for a better approach.